source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -210,6 +227,16 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
   Stats::StatName empty_stat_name;
 
   if (!response->dynamic_metadata.fields().empty()) {
+    // Add duration of call to dynamic metadata if applicable
+    StreamInfo::TimingUtility timingUtility(decoder_callbacks_->streamInfo());
+    auto ext_authz_duration = timeSince(start_time_);
+    if (ext_authz_duration.has_value()) {
```

Comment:
It seems that the if statement is true iff `start_time_` was set.
In this case, why not have the if statement (`if (start_time.has_value()) {`) here, and then set the number from the `time_since_epoch()` computation?

Reply:
Changes have been made. Good catch.